src/odb/src/db/dbITerm.cpp

```diff
@@ -666,4 +671,51 @@ void dbITerm::staSetVertexId(uint32_t id)
   iterm->_sta_vertex_id = id;
 }
 
+void dbITerm::setAccessPoint(dbMPin* pin, dbAccessPoint* ap)
+{
+  _dbITerm* iterm = (_dbITerm*) this;
+  if (ap != nullptr) {
+    iterm->aps_[pin->getImpl()->getOID()] = ap->getImpl()->getOID();
+    _dbAccessPoint* _ap = (_dbAccessPoint*) ap;
+    _ap->iterms_.push_back(iterm->getOID());
+  }
+  else
+    iterm->aps_[pin->getImpl()->getOID()] = dbId<_dbAccessPoint>();
+}
+
+std::vector<std::vector<dbAccessPoint*>> dbITerm::getAccessPoints() const
```

Comment:
I think it would be clearer to return a map of pin->aps rather than a vector with implied ordering.